doc/stdenv/stdenv.xml

```diff
@@ -2111,16 +2111,106 @@ postInstall = ''
       </para>
      </listitem>
     </varlistentry>
+
     <varlistentry>
      <term>
-      cmake
+      <anchor xml:id="cmake"/>
+       cmake
      </term>
      <listitem>
       <para>
-       Overrides the default configure phase to run the CMake command. By default, we use the Make generator of CMake. In addition, dependencies are added automatically to CMAKE_PREFIX_PATH so that packages are correctly detected by CMake. Some additional flags are passed in to give similar behavior to configure-based packages. You can disable this hook’s behavior by setting configurePhase to a custom value, or by setting dontUseCmakeConfigure. cmakeFlags controls flags passed only to CMake. By default, parallel building is enabled as CMake supports parallel building almost everywhere. When Ninja is also in use, CMake will detect that and use the ninja generator.
+       Overrides the default configure phase to run the CMake command. By default, we use the Make generator of CMake, however, if ninja is present, then the ninja generator will be used instead. In addition, dependencies are added automatically to CMAKE_PREFIX_PATH so that packages are correctly detected by CMake. By default, the cmake configurePhase can be thought of as:
+       <programlisting>
+mkdir build
+cd build
+cmake ..</programlisting>
```

Comment:
“Some additional flags are passed in to give similar behavior to configure-based packages.” This is meaningful, the default flags can break projects with poorly written build scripts (this is so common, I have rage-written https://github.com/jtojnar/cmake-snips).

“cmakeFlags controls flags passed only to CMake.”

We might want to show that here as well to show it is configurable?

Reply:
this was meant to give more of an intuition as to what the configure step was doing without any additional configuration.

I was afraid of adding more context, otherwise I would end up with:
```
    if [ -z "${dontUseCmakeBuildDir-}" ]; then
        mkdir -p build
        cd build
        cmakeDir=${cmakeDir:-..}
    fi

  cmake ${cmakeDir:-.} $cmakeFlags "${cmakeFlagsArray[@]}"
```

Reply:
I would be fine, with chaging the preceding line to something along the lines of "Excluding flags, the default configure step can be though doing:"

Reply:
Maybe just use `cmake .. $cmakeFlags` and commenting that `cmakeFlags` are various flags passed by the setup hook.

Reply:
that's fair